lib/scavenger_hunt/app/controllers/scavenger_hunt/answers_controller.rb

```diff
@@ -1,29 +1,31 @@
-# Make resources type-ahead-able
-require "pg_search"
-
-Resource.send :include, PgSearch
-Resource.send :pg_search_scope, :typeahead, against: :title, using: {
-  tsearch: {
-    normalization: 4,
-    prefix: true
-  }
-}
-
 class ScavengerHunt::AnswersController < ScavengerHunt::ApplicationController
-
-  TYPEAHEAD_LIMIT = 10
-
   before_action :find_game
   before_action :find_clue
 
   def correct
   end
 
   def create
-    @answer = @clue.answers.create!(answer_params)
+    # use the similar_text gem's .similar method to score how closely the player's input and the resource's title match (integer)
+    @match_score = "#{@clue.resource.title}".similar("#{params[:q]}")
+    puts "this is the match score #{@match_score}"
+
+    # setting a minimum match score of 40 allows the player some wiggle room for spelling errors
+    if @match_score >= 40
+      @resource = @clue.resource
+    else
+      @resource = nil
+    end
+
+    if @resource.present?
+      @answer = @clue.answers.create!(resource_id: @resource.id)
+    else
+      @answer = ScavengerHunt::Answer.new
+    end
```

Comment:
there may be an opportunity to use [find_or_create_by](https://api.rubyonrails.org/classes/ActiveRecord/Relation.html#method-i-find_or_create_by) here - which would potentially eliminate the need for this conditional